src/Analyzer.JsonRuleEngine/Converters/ExpressionConverter.cs

```diff
@@ -70,6 +81,10 @@ public override ExpressionDefinition ReadJson(JsonReader reader, Type objectType
                 {
                     return CreateExpressionDefinition<AllOfExpressionDefinition>(jsonObject, serializer);
                 }
+                else if (objectPropertyNames.Contains("anyOf", StringComparer.OrdinalIgnoreCase))
```

Comment:
@JohnathonMohr Should we have a file dedicated to all the string values of all the operators?

Reply:
(This is something I was going to update separately, but I'm not going to get to it soon.)

This can be simplified by not HashSets of property names for allOf and anyOf.  It's only needed for LeafExpression because that's where those operator names are located, and it's a convenient way of getting all possible operator names in one go.  It's overkill for structured expressions because the hash sets will only ever have a single string.

Reply:
I simplified it a bit.